src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -687,12 +673,23 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     // Return Value:
     // - the key chord that executes the given action
     // - nullptr if the action is not bound to a key chord
-    Control::KeyChord ActionMap::GetKeyBindingForAction(winrt::hstring cmdID) const
+    Control::KeyChord ActionMap::GetKeyBindingForAction(winrt::hstring cmdID)
     {
-        // Check our internal state.
-        if (const auto cmd{ _GetActionByID(cmdID) })
+        if (!_ResolvedKeyActionMapCache)
         {
-            return cmd.Keys();
+            _RefreshKeyBindingCaches();
+        }
+
+        // I dislike that we have to do an O(n) lookup every time we want to get the keybinding for an action -

```

Comment:
meh, isn't the only actual use of this `TabBase::_UpdateSwitchToTabKeyChord`[^1]? I'm thinking that it's probably fine if that's O(n). That's not the end of the world. 

[^1]: that may not be true after this whole stack of pr's

Reply:
shouldn't be the only place - 📝 the new tab menu should use something like this

Reply:
Yes, this is called one time _per profile_ during menu construction.

We should probably do better ;P

cascadia\TerminalApp\TerminalPage.cpp
828:            const auto settingsKeyChord{ actionMap.GetKeyBindingForAction(L"Terminal.OpenSettingsUI") };
850:            const auto commandPaletteKeyChord{ actionMap.GetKeyBindingForAction(L"Terminal.ToggleCommandPalette") };
1026:        const auto profileKeyChord{ _settings.ActionMap().GetKeyBindingForAction(id) };

Reply:
So we would probably need another map that is effectively a mirror of this one - it would be `Command` -> `array of keychords`, which would mean we are effectively storing the same data twice... Any objections or better ideas?

Reply:
You could map String -> one key chord (user key chord)